drivers/adc/adc_stm32.c

```diff
@@ -657,6 +657,13 @@ static int start_read(const struct device *dev,
 	 */
 	LL_ADC_SetChannelPreSelection(adc, channel);
 #endif
+#if defined(CONFIG_SOC_SERIES_STM32U5X)
+	/*
+	 * Each channel in the sequence must be previously enabled in PCSEL.
+	 * This register controls the analog switch integrated in the IO level.
+	 */
+	LL_ADC_SetChannelPreselection(adc, channel);
```

Comment:
Based on the reference manual RM0456, ADC4 doesn't have PCSEL register, and this function in stm32u5xx_ll_adc.h doesn't have a check for this.  Maybe is a good idea to add a check for ADC number here.